tests/arxml/ar4_component_test.py

```diff
@@ -273,20 +273,61 @@ def test_create_nvblock_software_component(self):
                                         writingPriority='LOW',
                                         checkStaticBlockId=False,
                                         autoValidationAtShutdown=False,
-                                        cyclicWritePeriod=0)
+                                        cyclicWritePeriod=0.0)

```

Comment:
Actually, I would prefer if this particular change wasn't necessary. It's a bit pedantic to force the developer to wrap the zero with a float(0) expression or having to type 0.0 explicitly.

You can implement an "implicit cast" or type upgrade in behavior.py right after line 502.

```python
self.cyclicWritePeriod = cyclicWritePeriod
if isinstance(self.cyclicWritePeriod, int):
    self.cyclicWritePeriod = float(self.cyclicWritePeriod)

if check_input:
       self.check()
 ...
```

The check will now accept None or float which is fine. The error text and docstring should say float is **expected** but we actually accept the int type in addition to the expected types. This simplifies things for the Python developer, especially for the special value 0 which is commonly used.

Reply:
Looks like a better solution.
I will update the pull request.